### DISCUSSION TITLE
Search in a table with shared_ptr

### DISCUSSION THREAD

**Discussion by Pantau123** [fnc12/sqlite_orm#1090] [General]
Hi i have a datamodel with shared_ptr as member.
I want to filter a single dataset with a special member.
Example:
```
struct MyModuleDetails
{
  MyModuleDetails()  
  {}

  uint32_t    m_u32Id{};
  std::shared_ptr<std::string> m_pstrSerialNumber ;
};

sqlite_orm::make_table( "MyModuleDB", 
                          sqlite_orm::make_column( "id", &MyModuleDetails::m_u32Id, sqlite_orm::primary_key() ), 
                          sqlite_orm::make_column( "serialNumber", &MyModuleDetails::m_pstrSerialNumber ));
```
How can i search the dataset for a specific serial number?

Bests

**Comment by fnc12** [comment]
Hi. What is `&MyModuleDetails::m_pstrSerialNumber`? Also you don't have `m_pstrVersion` mapped to your storage. Anyway `std::shared_ptr` must work out of box

**Reply by Pantau123** [reply]
Sorry, i have make a copy & paste failure...
I test it with a select command.

**Reply by fnc12** [reply]
ok. Anyway it must work out of box no matter you use `std::shared_ptr<std::string>` or just `std::string`
